src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java

```diff
@@ -833,24 +848,19 @@ private void httpResponseBegin()
         this.decoderState = this::decodeHttpBegin;
         this.responseState = ResponseState.BEFORE_HEADERS;
         this.acceptReplyWindowPadding = 0;
-        this.connectReplyWindowPadding = 0;      // TODO send half to avoid initial race
+        this.connectReplyWindowPadding = 0;
 
         final int connectReplyWindowCredit = factory.maximumHeadersSize - connectReplyWindowBudget;
 
-        if (connectReplyWindowCredit < 0)
-        {
-            throw new IllegalStateException("over-provisioned window");
-        }
-
         if (connectReplyWindowCredit > 0)
         {
+            this.connectReplyWindowBudget += connectReplyWindowCredit;
             factory.writer.doWindow(connectReplyThrottle, sourceId, connectReplyWindowCredit, connectReplyWindowPadding);
         }
 
         // TODO: Support HTTP/1.1 Pipelined Responses (may be buffered already)
-        this.connectReplyWindowBudget = factory.maximumHeadersSize;
-        this.connectReplyWindowBudgetAdjustment = -factory.maximumHeadersSize;
         this.contentRemaining = 0;
+        this.initialConnectReplyWindowDrained = false;
```

Comment:
This is not just initial connect reply window, since it also applies to connect reply window between requests when the underlying connection is reused.

The purpose is also to create alignment between the application and network windows, so probably should incorporate that into the naming as well.

Suggest renaming to `connectReplyWindowAligned` instead.